src/averageAges.js

```diff
@@ -15,31 +15,53 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let peopleCentury = [...people];
+
+  if (century !== null && typeof century === 'number') {
+    peopleCentury = people.filter(person =>
+      Math.ceil(person.died / 100) === century);
+  }
+
+  const men = peopleCentury.filter(person => person.sex === 'm');
```

Comment:
In both cases, you have almost the same logic.
You can use a ternary operator inside the filter callback to make it a bit cleaner